pytorch_lightning/trainer/logging.py

```diff
@@ -57,7 +55,12 @@ def log_metrics(self, metrics, grad_norm_dic, step=None):
         # turn all tensors to scalars
         scalar_metrics = self.metrics_to_scalars(metrics)
 
-        step = step if step is not None else self.global_step
+        if "step" in scalar_metrics and step is None:
+            step = scalar_metrics.pop("step")
```

Comment:
is `pop` the proper way, it removes it from the dictionary...

Reply:
That was intended. Because if `step` is not removed, user would additionally observe `y=x` plot and will be unsatisfied.